app/src/main/java/org/mozilla/fenix/home/recentbookmarks/controller/RecentBookmarksController.kt

```diff
@@ -0,0 +1,63 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.home.recentbookmarks.controller
+
+import androidx.navigation.NavController
+import mozilla.appservices.places.BookmarkRoot
+import mozilla.components.concept.storage.BookmarkNode
+import org.mozilla.fenix.BrowserDirection
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.nav
+import org.mozilla.fenix.home.HomeFragmentDirections
+import org.mozilla.fenix.home.recentbookmarks.interactor.RecentBookmarksInteractor
+
+/**
+ * An interface that handles the view manipulation of the recently saved bookmarks on the
+ * Home screen.
+ */
+interface RecentBookmarksController {
+
+    /**
+     * @see [RecentBookmarksInteractor.onRecentBookmarkClicked]
+     */
+    fun handleBookmarkClicked(bookmark: BookmarkNode)
+
+    /**
+     * @see [RecentBookmarksInteractor.onShowAllBookmarksClicked]
+     */
+    fun handleShowAllBookmarksClicked()
+}
+
+/**
+ * The default implementation of [RecentBookmarksController].
+ */
+class DefaultRecentBookmarksController(
+    private val activity: HomeActivity,
+    private val navController: NavController
+) : RecentBookmarksController {
+
+    /**
+     * @see [RecentBookmarksController.handleBookmarkClicked]
+     */
```

Comment:
You can probably remove these kDoc since it is expected that you can infer them from the interface.